assets/r/main.R

```diff
@@ -153,3 +153,22 @@ theme_rattle <- function(base_size = 11, base_family = "") {
 # theme_rattle <- theme_economist
 
 theme_default <- theme_rattle
+
+# Check if a variable is a factor (including ordered factors) and has more than 20 levels.
+
+is_large_factor <- function(x, MAXFACTOR = 20) {
```

Comment:
This will be risky if main.R is run by rSource, since rSource will replace MAXFACTOR with 20?

Reply:
It can reduce the chance of raising bugs. Without setting ' MAXFACTOR = 20', the app keeps raising bugs of `Error: object 'is_large_factor' not found`

Reply:
And rSource does not run main.R. So `MAXFACTOR = 20` should be kept.